src/ansys/fluent/core/services/datamodel_se.py

```diff
@@ -886,9 +886,15 @@ def set_state(self, state: Optional[Any] = None, **kwargs) -> None:
 
         Raises
         ------
+        TypeError
+            If no arguments are provided to set_state.
         ReadOnlyObjectError
             If the object is read-only.
         """
+        if not state and not kwargs:
+            raise TypeError(
+                "set_state() requires either 'state' or 'kwargs' arguments."
```

Comment:
Do we know the use case for kwargs?

Reply:
Yes @seanpearsonuk. Updated the test.
"meshing.GlobalSettings.set_state(EnableOversetMeshing=False)"

Reply:
Do flobject-based APIs support it, too?

Reply:
Yes, https://github.com/ansys/pyfluent/blob/main/tests/test_flobject.py#L443

Reply:
Thank you @mkundu1 .